### PR TITLE
script to calculate voter registration conversion from each source

### PR DESCRIPTION
**Pull request by mjain-1**
#### What's this PR do? / What are the relevant tickets/trello cards?
https://trello.com/c/yyiHBI2I/1448-data-request-voter-lead-to-registration-conversion

### CODE REVIEW

File: KPIs/Scripts/VoterReg/leadConversion.R [hunk 1]
```diff
@@ -0,0 +1,69 @@
...
+  }
+  print(nrow(df))
+}
+
+# remove dosomething members
+df <- dplyr::filter(df, !grepl('*dosomething*', Email))
+print(nrow(df))
+
+# sorted by Email and Date
+df <- df[with(df, order(df$Email, df$Date)),]
```
Comment (sohaibhasan) [inline KPIs/Scripts/VoterReg/leadConversion.R]: dplyr has an easy peasy `arrange` function. You could replace the sequence below with a couple of piped executions of arrange, filter, left_join, and then the group by

File: KPIs/Scripts/VoterReg/leadConversion.R [hunk 2]
```diff
@@ -0,0 +1,69 @@
...
+  mutate(
+    year = year(created_at)
+  ) %>%
+  filter (
+    file == 'RockTheVote',
+    ds_vr_status %in% c('register-form', 'register-OVR')
+  )
+
+leads_fnames <- list.files('../../Downloads/', 'leads*', full.names=TRUE)
+
```
Comment (sohaibhasan) [inline KPIs/Scripts/VoterReg/leadConversion.R]: Mind adding a few comments here so it's easier to understand what's happening?

File: KPIs/Scripts/VoterReg/leadConversion.R (hunk 1 above)
Comment (mjain-1) [reply KPIs/Scripts/VoterReg/leadConversion.R]: ah ! okay updated to take better advantage of dplyr

File: KPIs/Scripts/VoterReg/leadConversion.R (hunk 2 above)
Comment (mjain-1) [reply KPIs/Scripts/VoterReg/leadConversion.R]: done!

**Pull request merged**